### PR TITLE
[risk=low][RW-7256] refactor RegistrationTasks into AccessModuleConfig for better AAR and DAR alignment

### PR DESCRIPTION
**Pull request by jmthibault79**
Code cleanup; no changes expected.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/homepage/data-access-requirements.tsx [hunk 1]
```diff
@@ -225,28 +227,55 @@ export const allModules: AccessModule[] = [
...
+
+// This needs to be a function, because we want it to evaluate at call time,
+// not at compile time, to ensure that we make use of the server config store.
+// This is important so that we can feature flag off registration tasks.
+//
+// Important: The completion criteria here needs to be kept synchronized with
+// the server-side logic, else users can get stuck on the DAR
+// without a next step:
+// https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java#L240-L272
+const getAccessModuleConfig = (): AccessModuleConfig[] => serverConfigStore.get().config ? ([
```
Comment (jmthibault79) [inline ui/src/app/pages/homepage/data-access-requirements.tsx:247]: here is what remains of the former RegistrationTasks

File: ui/src/app/pages/homepage/data-access-requirements.tsx [hunk 2]
```diff
@@ -225,28 +227,55 @@ export const allModules: AccessModule[] = [
   duccModule,
 ];
 
-const LoginGovTooltip = () => <TooltipTrigger
-    content={'For additional security, we require you to verify your identity by uploading a photo of your ID.'}>
-  <InfoIcon style={{margin: '0 0.3rem'}}/>
-</TooltipTrigger>;
-
-// TODO merge with RegistrationTasks after we remove RegistrationDashboard
```
Comment (jmthibault79) [inline ui/src/app/pages/homepage/data-access-requirements.tsx:233]: these + RegistrationTasks are now getAccessModuleConfig()

File: ui/src/app/pages/homepage/data-access-requirements.tsx [hunk 3]
```diff
@@ -225,28 +227,57 @@ export const allModules: AccessModule[] = [
...
-  // DUCC state is strictly internal to the RW
-  [AccessModule.DATAUSERCODEOFCONDUCT, () => {}],
-]);
+interface AccessModuleConfig {
+  isEnabledInEnvironment: boolean;  // either true or dependent on a feature flag
+  darLabel: JSX.Element;
+  externalSyncAction?: Function;
+}
+
+// This needs to be a function, because we want it to evaluate at call time,
```
Comment (jmthibault79) [inline ui/src/app/pages/homepage/data-access-requirements.tsx:236]: this comment is mostly copied from RegistrationTasks.  I'm not sure how much of it is still relevant.

**Review by calbach — APPROVED** [review]
thanks

File: ui/src/app/pages/access/access-renewal.tsx [hunk 4]
```diff
@@ -150,12 +151,19 @@ const ActionButton = (
 
 const BackArrow = withCircleBackground(() => <Arrow style={{height: 21, width: 18}}/>);
 
+interface CardProps {
+  step: number,
+  moduleStatus: AccessModuleStatus,
+  style: React.CSSProperties,
+  children,
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/pages/access/access-renewal.tsx:158]: nice, thanks

**Pull request merged**